src/Query/SelectQuery.php

```diff
@@ -138,8 +138,9 @@ protected function applyGroupBy(AsyncMysqlConnection $conn, dataset $data): data
    * Apply the HAVING clause to every (maybe grouped) row in the data set. Only return truthy results.
    */
   protected function applyHaving(AsyncMysqlConnection $conn, dataset $data): dataset {
-    if ($this->havingClause is nonnull) {
-      return Vec\filter($data, $row ==> (bool)$this->havingClause->evaluate($row, $conn));
+    $havingClause = $this->havingClause;
+    if ($havingClause is nonnull) {
+      return Vec\filter($data, $row ==> (bool)$havingClause->evaluate($row, $conn));
```

Comment:
HHVM 4.15-dev.2019-07-24 was not convinced of your nullcheck. This should be backwards compatible with HHVM4.0.